homeassistant/components/plugwise/gateway.py

```diff
@@ -64,95 +66,99 @@ async def async_setup_entry_gw(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     try:
         connected = await api.connect()
-
         if not connected:
-            _LOGGER.error("Unable to connect to Smile")
+            _LOGGER.error("Unable to connect to the Smile/Stretch")
             raise ConfigEntryNotReady
-
     except InvalidAuthentication:
         _LOGGER.error("Invalid username or Smile ID")
         return False
-
     except PlugwiseException as err:
-        _LOGGER.error("Error while communicating to device %s", api.smile_name)
+        _LOGGER.error("Error while communicating to the Smile/Stretch")
         raise ConfigEntryNotReady from err
-
     except asyncio.TimeoutError as err:
-        _LOGGER.error("Timeout while connecting to Smile %s", api.smile_name)
+        _LOGGER.error("Timeout while connecting to the Smile/Stretch")
         raise ConfigEntryNotReady from err
 
+    # Migrate to a valid unique_id when needed
+    if entry.unique_id is None:
+        if api.smile_version[0] != "1.8.0":
```

Comment:
Should this be a version comparison of a range instead of a string match?

Reply:
I need to double-check this, if I remember correctly these lines were needed in plugwise-beta, when migrating from Core-plugwise to the plugwise-beta custom_component. I think these lines can be left out.